src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php

```diff
@@ -0,0 +1,125 @@
+{
+    /** @var ApiClientInterface */
+    private $client;
+
+    /** @var ResponseCheckerInterface */
+    private $responseChecker;
+
+    /** @var IriConverterInterface */
+    private $iriConverter;
+
+    public function __construct(
+        ApiClientInterface $client,
+        ResponseCheckerInterface $responseChecker,
+        IriConverterInterface $iriConverter
+    ) {
+        $this->client = $client;
+        $this->responseChecker = $responseChecker;
+        $this->iriConverter = $iriConverter;
+    }
+
+    /**
+     * @When I browse shipments
+     */
+    public function iBrowseShipments(): void
+    {
+        $this->client->index();
+    }
+
+    /**
+     * @Then I should see( only) :count shipment(s) in the list
+     * @Then I should see a single shipment in the list
+     */
+    public function iShouldSeeCountShipmentsInList(int $count = 1): void
+    {
+        Assert::same($this->responseChecker->countCollectionItems($this->client->getLastResponse()), $count);
+    }
+
+    /**
+     * @Then /^I should see the shipment of (order "[^"]+") as "([^"]+)"$/
+     */
+    public function iShouldSeeTheShipmentOfOrderAs(OrderInterface $order, string $shippingState): void
+    {
+        Assert::true(
+            $this->responseChecker->hasItemWithKeysAndValues($this->client->getLastResponse(), [
+                'order' => $this->iriConverter->getIriFromItem($order),
+                'state' => strtolower($shippingState)]),
+            sprintf('Shipment for order %s with state %s does not exist', $order->getNumber(), $shippingState)
+        );
+    }
+
+    /**
+     * @Then /^I should see shipment for the ("[^"]+" order) as (\d+)(?:|st|nd|rd|th) in the list$/
+     */
+    public function iShouldSeeShipmentForTheOrderInTheList(OrderInterface $order, int $position): void
+    {
+        Assert::true(
+            $this->responseChecker->hasItemOnPositionWithValue(
+                $this->client->getLastResponse(),
+                --$position,
+                'order',
+                $this->iriConverter->getIriFromItem($order)),
+            sprintf('On position %s there is no shipment for order %s', $position, $order->getNumber())
+        );
+    }
+
+    /**
+     * @Then the shipment of the :orderNumber order should be :shippingState for :customer
+     * @Then the shipment of the :orderNumber order should be :shippingState for :customer in :channel channel
+     */
+    public function shipmentOfOrderShouldBe(
+        string $orderNumber,
+        string $shippingState,
+        CustomerInterface $customer,
+        ChannelInterface $channel = null
+    ): void {
+        $this->client->index();
+        
+        foreach ($this->responseChecker->getCollectionItemsWithValue($this->client->getLastResponse(), 'state', StringInflector::nameToLowercaseCode($shippingState)) as $shipment) {
+            $orderIri = $shipment['order'];
+            $this->client->showByIri($orderIri);
```

Comment:
We can take response directly for this call (https://github.com/Sylius/Sylius/pull/11220/files#diff-7cd67ba0c744465221bda1283b08928dR22)